core/src/main/java/org/elasticsearch/action/fieldstats/FieldStats.java

```diff
@@ -272,6 +335,9 @@ public final void writeTo(StreamOutput out) throws IOException {
      * otherwise <code>false</code> is returned
      */
     public boolean match(IndexConstraint constraint) {
+        if (hasMinMax == false) {
+            return false;
+        }
```

Comment:
This is the current behavior for fields that are not indexed so maybe better to just ignore these fields for now ? The index constraint works on indexed fields and always returns false on non-indexed fields, I can add a sentence like this in the docs if needed ?